### ISSUE TITLE
db query optimization

### ISSUE THREAD

**Issue by jamesuber** [OpenWaterAnalytics/epanet-rtx#12]
In response to

```
  vector< Point > somePoints;
  somePoints = someTimeSeries->points(1222869600,1222870200);
```

where someTimeSeries is backed by a MysqlPointRecord, following is the record of db queries, from the log file. somePoints contains exactly what I expected, but I was confused about why two queries are being conducted for each time, both a 'time >' and a 'time ='. Could this be a possible optimization opportunity.

130126 12:46:19   648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869600
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869600 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869660 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869720 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869780 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869840 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869900 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222869960 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222870020 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222870080 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time > 1222870140 LIMIT 1
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869660
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869720
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869780
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869840
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869900
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222869960
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222870020
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222870080
          648 Execute   SELECT time, value FROM points INNER JOIN timeseries_meta USING (series_id) WHERE name = 'L ReservoirCheckValve flow' AND time = 1222870140

**Comment by samhatchett** [comment]
related to unimplemented optimization in MysqlPointRecord. we will need to borrow some code from ScadaPointRecord (see the `hintAtRange` method). could be good to generalize further so there's not so much repeated code. maybe the DbPointRecord intermediate class could take on some of this functionality?

**Comment by samhatchett** [comment]
this should be fixed now (in 41203e5755268b557a51458c2b475837334f08d8). i think this was related to the default TimeSeries::Points() method implementation -- we need better Points() subclass overrides like in the new Resampler
